lib/ngx/re.lua

```diff
@@ -0,0 +1,174 @@
+-- I hereby assign copyright in this code to the lua-resty-core project,
+-- to be licensed under the same terms as the rest of the code.
+
+
+local ffi = require 'ffi'
+local bit = require "bit"
+local base = require "resty.core.base"
+local core_regex = require "resty.core.regex"
+
+
+local C = ffi.C
+local sub = string.sub
+local type = type
+local band = bit.band
+local new_tab = base.new_tab
+local tostring = tostring
+local tonumber = tonumber
+local math_max = math.max
+local math_min = math.min
+local re_match_compile = core_regex.re_match_compile
+local destroy_compiled_regex = core_regex.destroy_compiled_regex
+
+
+local FLAG_DFA           = 0x02
+
+
+local PCRE_ERROR_NOMATCH = -1
+
+
+local DEFAULT_SPLIT_RES_SIZE = 4
```

Comment:
I think these constants can be grouped together without using 2 blank lines in between.